starboard/nplb/player_test_fixture.cc

```diff
@@ -26,6 +27,152 @@ namespace nplb {
 using shared::starboard::player::video_dmp::VideoDmpReader;
 using testing::FakeGraphicsContextProvider;
 
+class SbPlayerTestFixture::GroupedSamplesIterator {
+ public:
+  explicit GroupedSamplesIterator(const GroupedSamples& grouped_samples)
+      : grouped_samples_(grouped_samples) {}
+
+  bool HasMoreAudio() const {
+    return audio_samples_index_ <
+           grouped_samples_.audio_samples_to_write_.size();
+  }
+
+  bool HasMoreVideo() const {
+    return video_samples_index_ <
+           grouped_samples_.video_samples_to_write_.size();
+  }
+
+  GroupedSamples::AudioSamplesDescriptor GetCurrentAudioSamplesToWrite() const {
+    SB_DCHECK(HasMoreAudio());
+    GroupedSamples::AudioSamplesDescriptor descriptor =
+        grouped_samples_.audio_samples_to_write_[audio_samples_index_];
+    descriptor.start_index += current_written_audio_samples_;
+    descriptor.samples_count -= current_written_audio_samples_;
```

Comment:
Just to double check, assume |start_index| is 0, and |sample_count| is 100.  When |current_written_audio_samples_| is 0, the result is [0, 100).  Now `AdvanceAudio(1)` is called and |current_written_audio_samples_| is set to 1.  Then this function will return [1, 99).  Is this expected?

Reply:
Yes, that's right.

Reply:
As discussed offline, to have such a class handling the advancing logic is great, but it's not an iterator.  Let's add a TODO and address this in future.